src/Tasks/Microsoft.NET.Build.Tasks/targets/Microsoft.NET.Sdk.TargetingPackResolution.targets

```
@@ -154,6 +154,20 @@ Copyright (c) .NET Foundation. All rights reserved.
 
   </Target>
 
+  <Target Name="AddTransitiveFrameworkReferences" AfterTargets="ResolvePackageAssets"
+          Condition="'@(TransitiveFrameworkReferences)' != ''" >
+
+    <ItemGroup>
+      <!-- Deduplicate transitive framework references, and ignore any that were already directly
+           referenced-->
+      <_TransitiveFrameworkReferenceToResolve Include="@(TransitiveFrameworkReferences->Distinct())"/>
+      <_TransitiveFrameworkReferenceToResolve Remove="@(FrameworkReference)" />
+
+      <FrameworkReference Include="@(_TransitiveFrameworkReferenceToResolve)" />
```

After deduping in C#, I think this could be:

```suggestion
      <FrameworkReference Include="@(TransitiveFrameworkReferences)"  Exclude="@(FrameworkReference)" />
```